Make emoji filtering case insensitive

ZhaoZuohong opened the pull request:
Currently when filtering emoji, input strings are lowered first. Then they are compared with descriptions of those emoji. However, there are times that the description of a emoji contains upper-case characters. This happens a lot with national flags, since the first letter of the name of a country is usually in upper-case. For example, when searching for *china*, there is no chance that *China* will appear.

I just made a slight modification: when doing emoji filtering, compare lowered input letters with lowered emoji descriptions.

![image](https://user-images.githubusercontent.com/34163622/204235903-4355ff3a-6a8f-40ff-9f89-174de2693b35.png)

kernc commented:
Nice catch! 😊

Since only country flag emojis appear to be affected and since lowercase/insensitive seems the preferred friendly form, we might as well `.lower()` just once at emoji load time:
https://github.com/efck-chat-keyboard/efck/blob/c0301d5ffde4f2039f8dc9177e9c21983589b7b5/efck/emoji.py#L70-L96
This would then show countries names in lowercase, but should be much faster at painting while scrolling, and the Delegate will then also pick the matching text in bold (**china** in "\*china\*").

We should add an additional `re.I` here to make it consistent:
https://github.com/efck-chat-keyboard/efck/blob/c0301d5ffde4f2039f8dc9177e9c21983589b7b5/efck/tabs/emoji.py#L148

Thanks!

ZhaoZuohong commented:
I reverted previous changes, and lowered `name` in `efck/efck/emoji.py`. Now highlighting works:

![image](https://user-images.githubusercontent.com/34163622/205307086-8360926f-1653-4ca7-b701-8a6c3c652a99.png)

I have also added parameter `flags=re.I` in `efck/efck/tabs/emoji.py`'s `re.compile()`, although I think it is unnecessary, since input words and emoji description texts were lowered before.

kernc commented:
Thanks! :cake:
